### PR TITLE
[TAS-2199] 🚸 Improve affiliation link UX

### PR DESCRIPTION
**Pull request by nwingt**
Capture-2024-09-13-151233
Capture-2024-09-13-150345

### CODE REVIEW

File: stores/user.ts [hunk 1]
```diff
@@ -53,11 +60,39 @@ export const useUserStore = defineStore('user', () => {
     }
   }
 
+  async function fetchLikerInfo ({ nocache = false } = {}) {
```
Comment (williamchong) [inline stores/user.ts:63]:
```suggestion
  async function fetchLikerProfile ({ nocache = false } = {}) {
```
Or any other name that we won't confuse this with `fetchLikerInfoById`

File: stores/static-data.ts
Comment (williamchong) [inline stores/static-data.ts:1]: other data like `nft` and `collection` data are `static` too, maybe just call this store `liker` or sth?

File: stores/static-data.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      likerUserInfoMap.value[likerId] = userInfo
+      return userInfo
+    } catch {
+      return null
+    }
+  }
+
+  async function lazyFetchChannelInfoById (channelId: string) {
+    const likerId = convertChannelIdToLikerId(channelId)
+    const result = await lazyFetchLikerUserInfoById(likerId)
```
Comment (williamchong) [inline stores/static-data.ts:40]: Can also validate if stripe connect exists using `fetchStripeConnectStatus`

File: utils/index.ts [hunk 3]
```diff
@@ -126,25 +127,29 @@ export function getPurchaseLink ({
...
     const url = new URL(customLink)
-    Object.entries(payload).forEach(([key, value]) => {
+    Object.entries(query).forEach(([key, value]) => {
       url.searchParams.set(key, value)
     })
     return url.toString()
   }
+  if (isForQRCode && queryInput?.utm_medium) {
+    query.utm_medium = `${queryInput.utm_medium}-qr`
+  }
```
Comment (williamchong) [inline utils/index.ts:149]: if `(!queryInput?.utm_medium)` then `utm_medium` should set to `qrcode`?

File: stores/liker.ts [hunk 4]
```diff
@@ -1,40 +1,78 @@
 import { defineStore } from 'pinia'
 
-import { FetchLikerInfoByIdResult } from 'utils/api'
+import { FetchLikerInfoResult } from 'utils/api'
 
 export const useLikerStore = defineStore('liker', () => {
-  const likerUserInfoMap = ref({} as Record<string, FetchLikerInfoByIdResult>)
+  const likerInfoMapById = ref({} as Record<string, FetchLikerInfoResult>)
+  const likerInfoMapByWallet = ref({} as Record<string, FetchLikerInfoResult>)
```
Comment (williamchong) [inline stores/liker.ts:7]: Can either of these be a computed value of the other one?

Comment (nwingt) [reply stores/liker.ts:7]: How about keeping a map of likerId to wallet?

Comment (williamchong) [reply stores/liker.ts:7]: Can, or compute a map from another map using Object.entries and .reduce or sth

Comment (nwingt) [reply stores/liker.ts]: Done

image

**Review by williamchong — APPROVED** [review]

**Pull request merged**